README.md

```diff
@@ -1,2 +1,232 @@
+
+interface FormValues {
+  name: string;
+  email: string;
+}
+
+const FormExample = () => {
+  const [formModel, setFormModel] = useState<FormValues| null>(null);
+  const form = useRef<RxControlGroup<FormValues> | null>(null);
+
+  useEffect(() => {
+    // Create the form
+    form.current = FormFactory.group<FormValues>({
+      name: FormFactory.control(''),
+      email: FormFactory.control(''),
+    })
+
+    // Set the form model
+    setFormModel(form.current.value);
+  }, [])
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (form.current!.isValid()) {
+      // Do something with the form values
+      console.log(form.current!.value);
+    }
+  }
+
+  const refresh = () => {
+    // Refresh the form model
+    setFormModel(form.current!.value);
+  }
+
+  // We only render the form when the form model is ready
+  if (!formModel) {
+    return null;
+  }
+
+  return (<form onSubmit={handleSubmit} onChange={() => refresh()}>
+      <input type="text" onChange={(e) => form.current!.get('name')!.value = e.target.value} />
+      <input type="email" onChange={(e) => form.current!.get('email')!.value = e.target.value} />
+      <button type="submit">Submit</button>
+      <p>{JSON.stringify(formModel)}</p>
+  </form>)
+}
+```
+
+We can also use the `useForm` hook to create the form, and simplify the code.
+
+```tsx
+function useForm<T>(formGroup: RxControlGroup<T>): [
+  RxControlGroup<T> | null,
+  T | null,
+  (form: RxControlGroup<T> | undefined) => void
+] {
+  const [model, setModel] = useState<T | null>(null);
+  const formRef: MutableRefObject<RxControlGroup<T> | null> = useRef<RxControlGroup<T>>(null);
+
+  useLayoutEffect(() => {
+    if(!formGroup || formRef.current === formGroup) return;
+    formRef.current = formGroup;
+    setModel(formRef.current.value);
+  }, [formGroup]);
+
+  const onFormChange = (form: RxControlGroup<T> | undefined) => {
+    // add a debounce to avoid multiple renders
+    if(form) setModel(form.value);
+  }
+
+  return [ formRef.current, model, onFormChange ];
+}
+```
+
+So the previous example can be simplified to:
+
+```tsx
+import { useEffect, useRef, useState } from 'react';
+import { FormFactory } from '@kavehome/forms';
+import { RxControlGroup } from '@kavehome/forms/dist/types';
+import { useForm } from '.../useForm';
+
+interface FormValues {
+  name: string;
+  email: string;
+}
+
+function FormExample() {
+  const initForm = useMemo(() => FormFactory.group<FormValues>({
+    name: FormFactory.control(''),
+    email: FormFactory.control(''),
+  }), []);
+  const [form, model, onFormChange] = useForm<FormValues>(initForm);
+
+  if (!form) {
+    return null;
+  }
+
+  return (<form onChange={() => onFormChange(form)}>
+      <input type="text" onChange={(e) => form!.get('name')!.value = e.target.value} />
+      <input type="email" onChange={(e) => form!.get('email')!.value = e.target.value} />
```

Comment:
it is really necessary to add the `!` right after the `form`? In line 123 there is an if to validate form is defined